File: DataPath/Lua/C2L/LuaCallEnum.lua
--CS.命名空间.枚举名.枚举值
print(CS.TestEnum.Lol)
print(CS.TestEnum.Dota)

--转换获得枚举值
--注意CastFrom前面有两个下划线
print(CS.TestEnum.__CastFrom(0))
print(CS.TestEnum.__CastFrom("Dota"))

File: C2L/LuaCallEnum.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
public enum TestEnum
{
    Lol,
    Dota,
}
public class LuaCallEnum : MonoBehaviour
{
    void Start()
    {
        xLuaEnv.Instance.Dostring("require('C2L/LuaCallEnum')");
    }

    private void OnDestroy()
    {
        xLuaEnv.Instance.Free();
    }


}
